### PR TITLE
Add type hints to index.py

### PR DESCRIPTION
**Pull request by KoStehner**
Hello,

I added annotations to `tables/index.py`.

Further, there is small annotation fix in `tables/indexes.py`.

Best regards
Ko

### CODE REVIEW

File: tables/index.py [hunk 1]
```diff
@@ -1130,8 +1145,14 @@ def cleanup_temp(self):
         # ...but the memory data cache is dirty now
         self.dirtycache = True
 
-    def get_neworder(self, neworder, src_disk, tmp_disk,
-                     lastrow, nslices, offset, dtype):
+    def get_neworder(self,
+                     neworder: np.ndarray,
+                     src_disk: EArray,
+                     tmp_disk: CArray,
```
Comment (avalentino) [inline tables/index.py:1151]: Maybe we could use more generic types for the two parameters above.
in principle anything supporting indexing should work.
I would suggest to use "Array"

**Review by avalentino — CHANGES_REQUESTED** [review]

File: tables/index.py (hunk 1 above)
Comment (KoStehner) [reply tables/index.py]: Thanks for the feedback. Changed both annotations to `Array`.

Comment (avalentino) [reply tables/index.py]: @KoStehner there is some issue with the CI
I think that you need to use quotes: `Array` --> `"Array"`

Comment (KoStehner) [reply tables/index.py]: You are right, My bad. I fixed it

**Review by avalentino — APPROVED** [review]

**Comment by avalentino** [comment]
Thanks @KoStehner

**Pull request merged**